src/foam/u2/ActionView.js

```diff
@@ -401,17 +416,35 @@ foam.CLASS({
       this.addClass(this.myClass(this.action.name));
     },
 
-    function addContent() {
+    async function addContent() {
       /** Add text or icon to button. **/
-      if ( this.icon ) {
-        this.start('img')
-          .style({ 'margin-right': this.label ? '4px' : 0 })
-          .attr('src', this.icon$)
-        .end();
+      var self = this;
+      var size = this.buttonStyle == this.buttonStyle.LINK ? '1em' : this.size.iconSize;
+      var iconStyle = { width: size, height: size, 'margin-right': this.label ? '4px' : '' }; 
+      
+      if ( this.themeIcon ) {
+        var indicator = this.themeIcon.clone(this).expandSVG();
+        this.start(this.HTMLView, { data: indicator }).attrs({ role: 'presentation' }).style(iconStyle).end();
+      } else if ( this.icon ) {
+        if ( this.icon.endsWith('.svg') ) {
+          var req  = this.HTTPRequest.create({
+            method: 'GET',
+            path: this.icon
+          });
+          await req.send().then(function(payload) {
```

Comment:
Do we need the 'await'?

Reply:
Yes, The label is supposed to be added after the icon in the DOM and without `await` that order would be flipped.

Reply:
That isn't going to be obvious to future maintainers. If we want it to be added after, couldn't we just add it after to avoid confusion?

Reply:
We do add it after on line 452. The await is only required for this one case where we generate an HTTP request and we want the promise to be resolved before the next part of the function is executed